lib/PlaidClient.js

```diff
@@ -214,7 +214,13 @@ Client.prototype.getTransactions =
 
 // getAllTransactions(String, Date, Date, Object?, Function)
 Client.prototype.getAllTransactions =
-  function(access_token, start_date, end_date, cb) {
+  function(access_token, start_date, end_date, options, cb) {
+    // juggle arguments
+    if (typeof options === 'function') {
+      cb = options;
+      options = {};
+    }
```

Comment:
I think we should also default `options` when it is undefined... its sort of strange that

`R.merge(undefined, { foo: 'bar' })` results in `{ foo: 'bar' }`

I would rather not rely on this behavior.

Reply:
I don't see the need to do that, if the user specifies `options` (options != function), then a valid argument should be provided (an object) and it shouldn't ever be undefined. You could argue you could be more strict about it and force an object, but that is pretty uncommon in js libs in my experience and shouldn't be necessary.

Reply:
Since options is optional, it is possible to call the function as a promise like:

```typescript
getAllTransactions(access_token, start_date, end_date).then(data => { // do something });
```

I think in this case it can be undefined.

Reply:
Oh I see, wasn't thinking promises, was assuming there'd be a callback, will fix.